core/src/main/java/fj/data/List.java

```diff
@@ -609,6 +619,119 @@ public final void foreachDoEffect(final Effect1<A> f) {
+
+  public <B> Promise<List<B>> traversePromise(final F<A, Promise<B>> f) {
+    return foldRight(
+        (a, acc) -> f.f(a).bind(b -> acc.fmap(bs -> bs.cons(b))),
+        Promise.promise(Strategy.idStrategy(), p(List.<B> nil())));
+  }
+
+  public <B> List<List<B>> traverseList(final F<A, List<B>> f) {
+    return foldRight(
+        (a, acc) -> f.f(a).bind(b -> acc.map(bs -> bs.cons(b))),
+        List.single(List.<B> nil()));
+  }
+
+  public <E, B> Validation<E, List<B>> traverseValidation(final F<A, Validation<E, B>> f) {
+    return foldRight(
+        (a, acc) -> f.f(a).bind(b -> acc.map(bs -> bs.cons(b))),
+        Validation.success(List.<B> nil()));
+  }
+
+  public <B> V2<List<B>> traverseV2(final F<A, V2<B>> f) {
+    return foldRight(
+        (a, acc) -> acc.apply(f.f(a).<F<List<B>, List<B>>> map(e -> es -> es.cons(e))),
+        v(List.<B> nil(), List.<B> nil()));
+  }
+
+  /**
+   * polymorphic traversal
+   */
+  public static <A, B> PTraversal<List<A>, List<B>, A, B> _pTraversal() {
+    return new PTraversal<List<A>, List<B>, A, B>() {
+
+      @Override
+      public <C> F<List<A>, F<C, List<B>>> modifyFunctionF(F<A, F<C, B>> f) {
+        return l -> l.traverseF(f);
+      }
+
+      @Override
+      public <L> F<List<A>, Either<L, List<B>>> modifyEitherF(F<A, Either<L, B>> f) {
+        return l -> l.traverseEither(f);
+      }
+
+      @Override
+      public F<List<A>, IO<List<B>>> modifyIOF(F<A, IO<B>> f) {
+        return l -> l.traverseIO(f);
+      }
+
+      @Override
+      public F<List<A>, Trampoline<List<B>>> modifyTrampolineF(F<A, Trampoline<B>> f) {
+        return l -> l.traverseTrampoline(f);
+      }
+
+      @Override
+      public F<List<A>, Promise<List<B>>> modifyPromiseF(F<A, Promise<B>> f) {
+        return l -> l.traversePromise(f);
+      }
+
+      @Override
+      public F<List<A>, List<List<B>>> modifyListF(F<A, List<B>> f) {
+        return l -> l.traverseList(f);
+      }
+
+      @Override
+      public F<List<A>, Option<List<B>>> modifyOptionF(F<A, Option<B>> f) {
+        return l -> l.traverseOption(f);
+      }
+
+      @Override
+      public F<List<A>, Stream<List<B>>> modifyStreamF(F<A, Stream<B>> f) {
+        return l -> l.traverseStream(f);
+      }
+
+      @Override
+      public F<List<A>, P1<List<B>>> modifyP1F(F<A, P1<B>> f) {
+        return l -> l.traverseP1(f);
+      }
+
+      @Override
+      public <E> F<List<A>, Validation<E, List<B>>> modifyValidationF(F<A, Validation<E, B>> f) {
+        return l -> l.traverseValidation(f);
+      }
+
+      @Override
+      public F<List<A>, V2<List<B>>> modifyV2F(F<A, V2<B>> f) {
+        return l -> l.traverseV2(f);
+      }
+
+      @Override
+      public <M> F<List<A>, M> foldMap(Monoid<M> monoid, F<A, M> f) {
+        return l -> monoid.sumLeft(l.map(f));
+      }
+    };
+  }
+
+  /**
+   * monomorphic traversal
+   */
+  public static <A> Traversal<List<A>, A> _traversal() {
+    return new Traversal<>(_pTraversal());
+  }
+
```

Comment:
I prefer `lensTraversal` to `_traversal` for this method.  I don't like relying on using underscore to indicate the returned value is a lens.  Take `_head` vs `lensHead` for example.  

However, take note that I am a beginner user of lens.  

This would mean the `P2._1Lens` would be renamed to `lens_1`.  The polymorphic version could be `lensp_1` or `plens_1`.

Your thoughts?